### PR TITLE
enh(gorgone) Update config form to manage gorgone options 

### PR DESCRIPTION
**Pull request by loiclau**
## Description

Update admin > parameters form to manage gorgone options

continuation of https://github.com/centreon/centreon/pull/8184 for others form

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/install/insertTopology.sql [hunk 1]
```diff
@@ -79,7 +79,7 @@ INSERT INTO `topology` (`topology_id`, `topology_name`, `topology_parent`, `topo
...
-INSERT INTO `topology` (`topology_id`, `topology_name`, `topology_parent`, `topology_page`, `topology_order`, `topology_group`, `topology_url`, `topology_url_opt`, `topology_popup`, `topology_modules`, `topology_show`, `topology_style_class`, `topology_style_id`, `topology_OnClick`, `readonly`) VALUES (128,'CentCore',501,50117,25,1,'./include/Administration/parameters/parameters.php','&o=centcore','0','0','1',NULL,NULL,NULL,'1');
+INSERT INTO `topology` (`topology_id`, `topology_name`, `topology_parent`, `topology_page`, `topology_order`, `topology_group`, `topology_url`, `topology_url_opt`, `topology_popup`, `topology_modules`, `topology_show`, `topology_style_class`, `topology_style_id`, `topology_OnClick`, `readonly`) VALUES (128,'CentCore',501,50117,25,1,'./include/Administration/parameters/parameters.php','&o=gorgone','0','0','1',NULL,NULL,NULL,'1');
```
Comment (loiclau) [inline www/install/insertTopology.sql]:
```suggestion
INSERT INTO `topology` (`topology_id`, `topology_name`, `topology_parent`, `topology_page`, `topology_order`, `topology_group`, `topology_url`, `topology_url_opt`, `topology_popup`, `topology_modules`, `topology_show`, `topology_style_class`, `topology_style_id`, `topology_OnClick`, `readonly`) VALUES (128,'Gorgone',501,50117,25,1,'./include/Administration/parameters/parameters.php','&o=gorgone','0','0','1',NULL,NULL,NULL,'1');
```

File: doc/en/administration_guide/parameters.rst [hunk 2]
```diff
@@ -252,5 +252,5 @@ This part can be used to configure the enabling of the logging of activity on Ce
 * **RRDTool debug** box enables logging of the RRDTool graph engine debugging
 * **LDAP User Import debug** box enables logging of debugging of the import of LDAP users
 * **SQL debug** box enables the logging of SQL requests executed by the Centreon interface
-* **Centcore Engine debug** box enables logging of Centcore process debugging
+* **Centreon Gorgone debug** box enables logging of Centcore process debugging
```
Comment (loiclau) [inline doc/en/administration_guide/parameters.rst]:
```suggestion
* **Centreon Gorgone debug** box enables logging of Gorgone process debugging
```

File: www/install/insertBaseConf.sql [hunk 3]
```diff
@@ -140,7 +140,7 @@ INSERT INTO `options` (`key`, `value`) VALUES
 ('selectPaginationSize', 60),
 ('display_downtime_chart','0'),
 ('display_comment_chart','0'),
-('centcore_illegal_characters', '`'),
+('gorgoneillegal_characters', '`'),
```
Comment (loiclau) [inline www/install/insertBaseConf.sql]:
```suggestion
('gorgone_illegal_characters', '`'),
```

File: lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 4]
```diff
@@ -4590,8 +4590,8 @@ msgid "LDAP User Import debug"
 msgstr "Depuración de las importaciones de usuarios LDAP"
 
 #: centreon-web/www/include/Administration/parameters/debug/form.php:62
-msgid "Centcore Engine debug"
-msgstr "Depurando el motor de Centcore"
+msgid "Centreon Gorgone debug"
+msgstr "Depurando de Centreon Gorgone "
```
Comment (sc979) [inline lang/es_ES.UTF-8/LC_MESSAGES/messages.po]: are you sure for the "de" ?

File: www/include/Administration/parameters/DB-Func.php [hunk 5]
```diff
@@ -290,17 +290,18 @@ function updateCentcoreConfigData($db, $form, $centreon)
...
     );
     updateOption($db, "enable_logs_sync", isset($ret["enable_logs_sync"]) && $ret['enable_logs_sync'] ? 1 : 0);
+
     updateOption($db, "enable_broker_stats", isset($ret["enable_broker_stats"]) && $ret['enable_broker_stats'] ? 1 : 0);
     updateOption(
         $db,
-        "centcore_cmd_timeout",
-        isset($ret["centcore_cmd_timeout"]) && $ret['centcore_cmd_timeout'] ? $ret['centcore_cmd_timeout'] : 0
+        "gorgone_cmd_timeout",
+        isset($ret["gorgone_cmd_timeout"]) && $ret['gorgone_cmd_timeout'] ? $ret['gorgone_cmd_timeout'] : 0
```
Comment (sc979) [inline www/include/Administration/parameters/DB-Func.php]: $ret["gorgone_cmd_timeout"] ?? 0

File: www/include/Administration/parameters/DB-Func.php [hunk 6]
```diff
@@ -290,17 +290,18 @@ function updateCentcoreConfigData($db, $form, $centreon)
...
     );
     updateOption(
         $db,
-        "centcore_illegal_characters",
-        isset($ret["centcore_illegal_characters"]) && $ret['centcore_illegal_characters']
-            ? $ret['centcore_illegal_characters']
+        "gorgone_illegal_characters",
+        isset($ret["gorgone_illegal_characters"]) && $ret['gorgone_illegal_characters']
```
Comment (sc979) [inline www/include/Administration/parameters/DB-Func.php]: $ret["gorgone_illegal_characters"] ?? ""

File: www/include/Administration/parameters/api/help.php [hunk 7]
```diff
@@ -12,9 +12,9 @@
     'help',
     'Enable Monitoring Engine Logs synchronisation between poller and Central Server operated by Centcore'
 );
-$help['tip_centcore_cmd_timeout'] = dgettext(
+$help['tip_gorgone_cmd_timeout'] = dgettext(
```
Comment (sc979) [inline www/include/Administration/parameters/api/help.php]: does this option currently exist in gorgone ?

File: lang/es_ES.UTF-8/LC_MESSAGES/messages.po (hunk 4 above)
Comment (loiclau) [reply lang/es_ES.UTF-8/LC_MESSAGES/messages.po]: no it's the 'de' of:  motor **de** Centcore

File: www/include/Administration/parameters/api/help.php (hunk 7 above)
Comment (cgagnaire) [reply www/include/Administration/parameters/api/help.php]: Well actually Gorgone has lots of different timeouts on all his modules.
But this one here can/will be use when calling for an action like external commands.
So yes for now it's interesting to have a timeout set somewhere on the UI side.

The help sentence must be changed though.
Something like : "Timeout value in seconds. Used to make actions calls timeout."

File: www/include/Administration/parameters/api/api.php [hunk 8]
```diff
@@ -57,7 +57,7 @@
  * Centcore Options
  */
 $form->addElement('checkbox', 'enable_broker_stats', _("Enable Broker Statistics Collection"));
-$form->addElement('text', 'centcore_cmd_timeout', _("Timeout value for Centcore commands"), $attrsText2);
+$form->addElement('text', 'centcore_cmd_timeout', _("Timeout value for Gorgone commands"), $attrsText2);
```
Comment (cgagnaire) [inline www/include/Administration/parameters/api/api.php]:
```suggestion
$form->addElement('text', 'gorgone_cmd_timeout', _("Timeout value for Gorgone commands"), $attrsText2);
```

File: www/include/Administration/parameters/api/help.php [hunk 9]
```diff
@@ -12,9 +12,9 @@
...
     'help',
-    "Timeout value in seconds. Used for freeing Centcore when it's stuck because"
+    "Timeout value in seconds. Used for freeing centreon-gorgone when it's stuck because"
```
Comment (cgagnaire) [inline www/include/Administration/parameters/api/help.php]:
```suggestion
    "Timeout value in seconds. Used to make actions calls timeout."
```

File: www/include/Administration/parameters/api/help.php [hunk 10]
```diff
@@ -12,9 +12,9 @@
...
     . " of blocking pipe files or SSH connections."
```
Comment (cgagnaire) [inline www/include/Administration/parameters/api/help.php]:
```suggestion
```

File: www/include/Administration/parameters/api/help.php (hunk 9 above)
Comment (loiclau) [inline www/include/Administration/parameters/api/help.php]:
```suggestion
    "Timeout value in seconds. Used for freeing centreon-gorgone when it's stuck"
```

File: www/include/Administration/parameters/gorgone/help.php [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+    'help',
+    'Enable Perfdata synchronisation between poller and Central Server operated by Gorgone'
+);
+$help['tip_enable_logs_sync'] = dgettext(
+    'help',
+    'Enable Monitoring Engine Logs synchronisation between poller and Central Server operated by Gorgone'
+);
+$help['tip_gorgone_cmd_timeout'] = dgettext(
+    'help',
+    "Timeout value in seconds. Used for freeing centreon-gorgone when it's stuck"
```
Comment (cgagnaire) [inline www/include/Administration/parameters/gorgone/help.php]:
```suggestion
    "Timeout value in seconds. Used to make actions calls timeout."
```

File: www/include/Administration/parameters/gorgone/help.php [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+    . " because of blocking pipe files or SSH connections."
```
Comment (cgagnaire) [inline www/include/Administration/parameters/gorgone/help.php]:
```suggestion
```

File: www/include/Administration/parameters/gorgone/help.php [hunk 13]
```diff
@@ -0,0 +1,30 @@
+<?php
+$help = array();
+
+/**
+ * Centcore Settings
```
Comment (cgagnaire) [inline www/include/Administration/parameters/gorgone/help.php]:
```suggestion
 * Gorgone Settings
```

File: www/include/Administration/parameters/gorgone/help.php [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+ */
+$help['tip_enable_perfdata_sync'] = dgettext(
+    'help',
+    'Enable Perfdata synchronisation between poller and Central Server operated by Gorgone'
+);
+$help['tip_enable_logs_sync'] = dgettext(
```
Comment (cgagnaire) [inline www/include/Administration/parameters/gorgone/help.php]: Not used anymore.

File: www/include/Administration/parameters/gorgone/help.php [hunk 15]
```diff
@@ -0,0 +1,30 @@
+<?php
+$help = array();
+
+/**
+ * Centcore Settings
+ */
+$help['tip_enable_perfdata_sync'] = dgettext(
```
Comment (cgagnaire) [inline www/include/Administration/parameters/gorgone/help.php]: Not used anymore.

**Pull request merged**